lib/sqlitex/sql_builder.ex

```diff
@@ -0,0 +1,74 @@
+defmodule Sqlitex.SqlBuilder do
+  @moduledoc """
+  This module contains functions for SQL creation. At the moment
+  it is only used for `CREATE TABLE` statements.
+  """
+  
+  # Supported table options
+  defp table_opt(:temporary), do: {:temp, "TEMP"}
+  defp table_opt(:temp), do: {:temp, "TEMP"}
+
+  defp table_opt({:primary_key, cols}) when is_list(cols) do
+    {:primary_key, ",PRIMARY KEY (" <> Enum.join(cols, ",") <> ")"}
+  end
+
+  defp table_opt({:primary_key, col}) when is_atom(col) do
+    {:primary_key, ",PRIMARY KEY (" <> Atom.to_string(col) <> ")"}
+  end
+
+  # Catch invalid / unsupported table options
+  defp table_opt(opt), do: raise "Unkown table option: #{IO.inspect(opt)}"
+
+  # Supported column options
+  defp column_opt(:primary_key), do: {:primary_key, "PRIMARY KEY"}
+  defp column_opt(:not_null), do: {:not_null, "NOT NULL"}
+  defp column_opt(:autoincrement), do: {:autoincrement, "AUTOINCREMENT"}
+
+  # Catch invalid / unsupported column options
+  defp column_opt(opt), do: raise "Unknown column option: #{IO.inspect(opt)}"
+
+  # Helper function that creates a dictionary of option names
+  # and their string representations
+  defp get_opts_dict(opts, opt) do
+    Enum.into(opts, %{}, &(opt.(&1)))
+  end
+
+  # Create the sql fragment for the column definitions from the
+  # passed keyword list
+  defp get_columns_block(cols) do
+    Enum.reduce(cols, "", fn(col, acc) ->
+      comma = if acc == "" do nil else "," end
```

Comment:
Correct me if I'm wrong, but it looks like you are trying to account for the comma separators between column definitions.  It might be cleaner and easier to `map` the columns to a list of individual definitions and then pipe that output to `Enum.join(", ")`.  For example:

```elixir
cols
|> Enum.map(fn
  {name, {type, constraints}} -> # blah
  {name, type} -> # blah
end)
|> Enum.join(", ")
```

The above code can be further simplified with [`Enum.map_join/3`](http://elixir-lang.org/docs/stable/elixir/Enum.html#map_join/3).